src/GNNGraphs/query.jl

```diff
@@ -273,6 +273,15 @@ or a vector.
                 Default `true`.
 
 """
+function Graphs.degree(g::GNNHeteroGraph, edge::Tuple{Symbol, Symbol, Symbol}; dir = :in)   
+    coo_rep = edge_index(g, edge)
+    if dir == :in
+        return [count(==(i), coo_rep[2]) for i in 1:g.num_nodes[edge[3]]]
```

Comment:
this is returning a cpu array even if the graph is on gpu. Also, it is inefficient.
You should use the utility function `_degree` defined below.